### PR TITLE
ROE-1772 Use entity name in the filing data

### PR DESCRIPTION
**Pull request by mattch1**
https://companieshouse.atlassian.net/browse/ROE-1772

Updated the filing data and tests with the new entity name field.
Removed existing entity name from the tests.

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/mocks/Mocks.java [hunk 1]
```diff
@@ -187,7 +188,6 @@ public static OverseasEntitySubmissionDto buildSubmissionDtoWithBoCorporateTrust
 
     private static EntityDto buildEntityDto() {
         EntityDto entityDto = new EntityDto();
-        entityDto.setName("Joe Bloggs Ltd");
```
Comment (mwestacott) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/mocks/Mocks.java:190]: Please revert this change - I've done this in my PR (https://github.com/companieshouse/overseas-entities-api/pull/157).

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingServiceTest.java [hunk 2]
```diff
@@ -170,11 +170,11 @@ void testFilingGenerationWhenSuccessfulWithoutTrustsAndWithIdentityChecks() thro
         assertEquals(FILING_KIND_OVERSEAS_ENTITY, filing.getKind());
         assertEquals(FILING_DESCRIPTION_IDENTIFIER, filing.getDescriptionIdentifier());
         assertEquals("Filing Description with registration date 26 March 2022", filing.getDescription());
+        assertEquals("Joe Bloggs Ltd", filing.getData().get("entity_name"));
         final PresenterDto presenterInFiling = (PresenterDto)filing.getData().get("presenter");
         assertEquals("Joe Bloggs", presenterInFiling.getFullName());
         assertEquals("user@domain.roe", presenterInFiling.getEmail());
         final EntityDto entityInFiling = ((EntityDto) filing.getData().get("entity"));
-        assertEquals("Joe Bloggs Ltd", entityInFiling.getName());
```
Comment (mwestacott) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingServiceTest.java:177]: Same with these - revert. You only need to worry about adding the new field to the top-level object and nothing related to removing the field from 'entity'.

Comment (mattch1) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingServiceTest.java]: Reverted

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/mocks/Mocks.java (hunk 1 above)
Comment (mattch1) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/mocks/Mocks.java]: Reverted

**Review by mwestacott — APPROVED** [review]
